server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableImpl.java

```diff
@@ -1388,11 +1388,12 @@ public synchronized void close() {
 
       closed = true;
 
-      // wait while internal jobs are running or external compactions are committing, but do not
-      // wait on external compactions that are running
+      // wait while internal jobs are running, external compactions are committing or the status of
+      // chops is active, but do not wait on external compactions that are running
       while (runningJobs.stream()
           .anyMatch(job -> !((CompactionExecutorIdImpl) job.getExecutor()).isExternalId())
-          || !externalCompactionsCommitting.isEmpty()) {
+          || !externalCompactionsCommitting.isEmpty()
+          || fileMgr.chopStatus != FileSelectionStatus.NOT_ACTIVE) {
```

Comment:
I think in the current code when a chop is actually marking the state will actually be NOT_ACTIVE.  Thinking we should introduce a new explicit state for MARKING and then we could do something like the following.

```suggestion
          || fileMgr.chopStatus == FileSelectionStatus.MARKING) {
```

To go with the above change could do something like the following in checkIfChopComplete()

```java
  private void checkifChopComplete(Set<StoredTabletFile> allFiles) {

    boolean completed = false;

    synchronized (this) {
      if (closed) {
       // if closed do not attempt transition to marking state
        return;
     }
      // when this returns true it means we transitioned to the MARKING state
      completed = fileMgr.finishChop(allFiles);
    }

    if (completed) {
      try {
        markChopped();
      } finally {
        synchronized (this) {
          // this will transition from MARKING to NOT_ACTIVE
          fileMgr.finishMarkingChop();
          this.notifyAll();
        }
      }

      TabletLogger.selected(getExtent(), CompactionKind.CHOP, Set.of());
    }
  }
```

In a [branch](https://github.com/keith-turner/accumulo/commits/accumulo-2199) I created for #2199 I have created a separate enum for chop status since chop only uses a subset of the file status states.  Adding an extra state that is only used for chop would make this separate enum even more useful.  For this PR could add the new state to the existing enum and I can fix it up later.

Reply:
> > For this PR could add the new state to the existing enum and I can fix it up later.
> 
> Okay I'll go ahead and do that here

@keith-turner, Actually do you think that the new state should just be added with the changes that will use it instead of adding it here before it is used anywhere? It seems like this PR could just be closed if there are plans to account for this later on.

Reply:
For the work on #2199 I would not be adding a new MARKING state, only a new enum specific to chop states.  Still need to make these changes. These changes could wait until the new enum is added and add the new state to that enum as that will be cleaner.

Reply:
@keith-turner, I tried incorporating your suggestions in a9c0f9b. Not too sure if they are what you had in mind.

Reply:
The changes look good. Need to see if the new state affects existing code.

Reply:
I wasn't too sure how to best test to see if these changes affect existing code so I ended up running all the ITs on this branch. Everything looks good (except a failing ConcurrentDeleteTableIT which is known flaky and also fails on main so is probably unrelated. Might create a ticket for this.).

I think these changes are ready for review @keith-turner